shopinvader/services/cart.py

```diff
@@ -68,7 +68,67 @@ def clear(self):
         cart = self._clear_cart(cart)
         return self._to_json(cart)
 
+    def _get_validator_cart_by_id_domain(self, value):
+        return [("typology", "=", "cart"), ("id", "=", value)]
+
+    def _get_line_copy_vals(self, line):
+        """
+        Prepare copy values to be passed to _add_item
+        :param line: sale.order.line
+        :return: dict
+        """
+        return {
+            "product_id": line.product_id.id,
+            "item_qty": line.product_uom_qty,
+        }
+
+    def copy(self, **params):
+        """
+        This service allows
+        :param params:
+        :return:
+        """
+        return self._copy(**params)
+
+    def _copy(self, **params):
+        """
+        Copy the cart given by id without the lines
+        They will be re-added
+        :return: dict/json
+        """
+        cart = self.env["sale.order"].search(
+            self._get_cart_by_id_domain(params.get("id"))
+        )
+        # Copy the existing cart
+        # Delete all lines and re-add them with 'shopinvader' flavour
+        new_cart = cart.copy({"order_line": False})
```

Comment:
don't we have to re-assign the cart then?

Reply:
What do you mean ?

Reply:
how do you use this option? I guess is for repeating an order from my account on the client.
Then, if I click "Duplicate" button I think a new cart must be created no?

Unless, I'm going beyond the scope of your change and you just want to re-place the same order as it is. " If there is an existing cart, it will be replaced" seems misleading in that case :thinking:

Reply:
Yes, the aim is to have a new cart at the end. I haven't done live tests done yet with a locomotive, but the _to_json method would return a 'set_session' that would update the actual cart.

Reply:
oh you are right, _to_json on cart service is "particular" :)